plugins/single_plugins/resize.cpp

```diff
@@ -349,9 +352,51 @@ class wayfire_resize : public wf::per_output_plugin_instance_t, public wf::point
             desired.height = std::max(desired.height, 1);
         }
 
-        view->toplevel()->pending().gravity  = calculate_gravity();
-        view->toplevel()->pending().geometry = desired;
-        wf::get_core().tx_manager->schedule_object(view->toplevel());
+        auto desired_unconstrained = desired;
+
+        if (min_size.width > 0)
+        {
+            desired.width =
+                std::max(min_size.width +
+                    (view->toplevel()->pending().margins.left + view->toplevel()->pending().margins.right),
+                    desired.width);
+        }
+
+        if (max_size.width > 0)
+        {
+            desired.width =
+                std::min(max_size.width -
```

Comment:
Why the subtraction here? max_size is what the client reports, but our full geometry can grow to client_max_size + margins from the decoration.